tools/snp.sh

```diff
@@ -164,6 +172,83 @@ verify_snp_host() {
   fi
 }
 
+verify_platform_snp_bit_status() {
+  if [ "$1" == "host" ]; then
+    # Get the host cpuid eax
+    local host_cpuid_eax
+    host_cpuid_eax=$(cpuid -1 -r -l 0x8000001f | grep -oE "eax=[[:alnum:]]+ " | cut -c5- | tr -d '[:space:]')
+
+    # Map all the computed host sev/snp bit values in a single associative array
+    declare -A actual_sev_snp_bit_status=(
+      [SME]=$((${host_cpuid_eax} & 1))
+      [SEV]=$((${host_cpuid_eax} & 2))
+      [SEV-ES]=$((${host_cpuid_eax} & 8))
+      [SNP]=$((${host_cpuid_eax} & 16))
+    )
+
+    # Map all the expected host sev/snp bit values in a single associative array
+    declare -A expected_sev_snp_bit_value=(
+      [SME]=1
+      [SEV]=2
+      [SEV-ES]=8
+      [SNP]=16
+      )
```

Comment:
You could just grab the specific bit for each of the features and that way you don't have to create two different arrays.